lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -352,6 +352,11 @@ def get_virtual_facts(self):
                         virtual_facts['virtualization_role'] = 'guest'
                         found_virt = True
 
+                if 'BHYVE' in out:
+                    virtual_facts['virtualization_type'] = 'bhyve'
+                    virtual_facts['virtualization_role'] = 'guest'
+                    found_virt = True
```

Comment:
I think we'll need something like:

```suggestion
                if 'BHYVE' in out:
                    guest_tech.add('bhyve')
                    if not found_virt:
                        virtual_facts['virtualization_type'] = 'bhyve'
                        virtual_facts['virtualization_role'] = 'guest'
                        found_virt = True
```

Basically: If there's nested or multiple virt techs at play and we found something earlier, go with that, but add bhyve to the list of guest techs no matter what.

Reply:
Will do.